src/inmanta/module.py

```diff
@@ -1366,18 +1415,20 @@ def use_module_cache() -> bool:
 
         if use_module_cache():
             return self.modules[module_name]
-        return self.load_module(module_name, install=install, allow_v1=allow_v1)
+        return self.load_module(module_name, allow_v1=allow_v1, install_v1=install_v1, install_v2=install_v2)
 
     def load_module_recursive(
```

Comment:
General questions: from a design perspective, we no longer have loading and install as a single function (i.e. `inmanta project install` does the install, `inmanta compile` the loading). Could we lose some complexity by decoupling these two things in the code flow as well? 

(e.g. if we know no install will happen, reloading of a V1 to V2 can also never happen)

Reply:
I think this would be difficult to accomplish as long as we still support v1 modules, that do have import-based installations. I actually started out that way but I ended up keeping it in `load_module_recursive`.